python-package/xgboost/collective.py

```diff
@@ -15,7 +16,41 @@
 LOGGER = logging.getLogger("[xgboost.collective]")
 
 
-def init(**args: Any) -> None:
+_ArgVals: TypeAlias = Optional[Union[int, str]]
+_Args: TypeAlias = Dict[str, _ArgVals]
+
+
+@dataclass
+class Config:
```

Comment:
Could we rename it make it more readable?

Reply:
And could we add json support in it?

Reply:
`xgboost.collective.Config` seems readable to me.

Reply:
Removed the converters. We can use the utilities from `dataclasses`, see the collective test for an exmaple.